lib/iex/lib/iex/introspection.ex

```diff
@@ -115,13 +126,37 @@ defmodule IEx.Introspection do
         end
         :ok
       else
-        :not_found
+        if has_callback?(mod, fun, arity) do
+          :behaviour_found
+        else
+          :not_found
+        end
```

Comment:
Not a big friend of this nested `if` statements either, but it seems to be a common pattern on this file, so if we're refactoring this we could go for all of the `h` functions here.

If that's the case (and I'd love to do it), since we have more like *nested* `if`s than *chained*, I would rather go for something more like a [railway pattern](http://www.zohaib.me/railway-programming-pattern-in-elixir/) here, probably without the `>>>` thing and just handling the different scenarios through pattern matching.
This would make the different parts more reusable.

It would be something like:

```elixir
defp h_mod_fun_arity(mod, fun, arity) when is_atom(mod) do
    {mod, fun, arity}
    |> h_get_docs
    |> h_find_doc
    |> h_handle_doc
  end
```

I think it's a rather big topic to tackle on this PR, but I'd love to do it on another one.

What do you think? Let's see what the other folks here have to say :)